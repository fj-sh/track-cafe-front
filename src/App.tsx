import React, { useState } from 'react'

import { menuData } from './lib/mocks/menu-data'
import { Menu } from './lib/types/menu'
import TrackCafeLayout from './components/TrackCafeLayout'

/**
 * React App
 * @constructor
 */
function App() {
  const [menus, setMenus] = useState(menuData)
  const updateMenus = (targetMenu: Menu) => {
    const updatedMenus = menus.map((menu) => {
      if (menu.id === targetMenu.id) {
        return { ...menu, numberOfOrders: menu.numberOfOrders + 1 }
      }
      return menu
    })
    setMenus(updatedMenus)
  }
  return (
    <div className="App">
      <TrackCafeLayout menus={menus} updateMenus={updateMenus} />
    </div>
  )
}

export default App
