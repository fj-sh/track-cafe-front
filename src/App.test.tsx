import React from 'react'
import { it } from 'vitest'
import { render } from '@testing-library/react'
import App from './App'

describe('Vitest を動かしてみる', () => {
  it('console.log', (ctx) => {
    console.log(ctx.meta.name)
  })
  it('正しくレンダリングされること', () => {
    const { container } = render(<App />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
