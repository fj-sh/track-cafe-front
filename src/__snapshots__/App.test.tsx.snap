// Vitest Snapshot v1

exports[`Vitest を動かしてみる > 正しくレンダリングされること 1`] = `
<div
  class="App"
>
  <div
    class="mx-10 my-10"
  >
    <div
      class="order-button"
    >
      <div
        class="menu"
      >
        <span
          class="menu-name"
        >
          コーヒー
        </span>
        <span
          class="menu-price"
        >
          480
           円
        </span>
      </div>
      <span
        class="badge"
      >
        0
      </span>
    </div>
  </div>
  <div
    class="mx-10 my-10"
  >
    <div
      class="order-button"
    >
      <div
        class="menu"
      >
        <span
          class="menu-name"
        >
          紅茶
        </span>
        <span
          class="menu-price"
        >
          280
           円
        </span>
      </div>
      <span
        class="badge"
      >
        0
      </span>
    </div>
  </div>
  <div
    class="mx-10 my-10"
  >
    <div
      class="order-button"
    >
      <div
        class="menu"
      >
        <span
          class="menu-name"
        >
          ミルク
        </span>
        <span
          class="menu-price"
        >
          180
           円
        </span>
      </div>
      <span
        class="badge"
      >
        0
      </span>
    </div>
  </div>
  <div
    class="mx-10 my-10"
  >
    <div
      class="order-button"
    >
      <div
        class="menu"
      >
        <span
          class="menu-name"
        >
          コーラ
        </span>
        <span
          class="menu-price"
        >
          190
           円
        </span>
      </div>
      <span
        class="badge"
      >
        0
      </span>
    </div>
  </div>
  <div
    class="mx-10 my-10"
  >
    <div
      class="order-button"
    >
      <div
        class="menu"
      >
        <span
          class="menu-name"
        >
          ビール
        </span>
        <span
          class="menu-price"
        >
          580
           円
        </span>
      </div>
      <span
        class="badge"
      >
        0
      </span>
    </div>
  </div>
</div>
`;
